app/helpers/general-date.js

```diff
@@ -3,7 +3,19 @@ import moment from 'moment';
 
 export function generalDate(params) {
   const timezone = params[1] || moment.tz.guess();
-  const format = params[2] || 'h:mm A, MMMM Do YYYY (z)';
+  let changeTo24 = false;
+
+  let format = params[2] || 'h:mm A, MMMM Do YYYY (z)';
+
+  if (moment(params[0]).tz(timezone).format('A') === 'AM' || moment(params[0]).tz(timezone).format('A') === 'PM') {
+    changeTo24 = true;
```

Comment:
I don't understand. What is this doing?

Reply:
I have checked for diff languages AM/PM is being converted by moment but not all the languages are supported in moment so the languages that are not supported gives AM/PM by default. So checking if they give AM/PM and if yes then converting format to 24 hr clock.

Reply:
But in English, we want AM/PM

Reply:
for that I need to see what can I do

Reply:
is there a way in which I can get which language is currently active or to how to get which language moment is using?

Reply:
It's better to maintain a set of locales for 12 hr/24 hr and switch based on detected locale